include/vsg/utils/GraphicsPipelineConfigurator.h

```diff
@@ -58,11 +58,13 @@ namespace vsg
         /// call after all the textures and buffers have been explictly assigned to add in texture/buffer descriptors that are enabled by default (define == "").
         bool assignDefaults();
 
-        /// deprecated. use enableBuffer() instead
+        [[deprecated("use enableBuffer()")]]
```

Comment:
Shouldn't this be enableDescriptor()?

W.r.t [[deprecated]], do you know whether doxygen can handle it in a sensible way?

Reply:
no. `GraphicsPipelineConfigurator::enableBuffer()` is still good in my opinion. see my previous comment posted at the same time as yours just now.

Reply:
I think enableBuffer() is more confusing that the original enableUniform().  Buffer has a distinct mean in VSG and Vulkan, it refers to a whole buffer not a subset that is can be as a uniform and storage.

enableDescriptor() is more generic than enableTexture()/enableUniform() but it does at least still map to DescriptorBuffer/DescriptorImage.  If we want to replace enableUniform()/enableTexture() then it would probably be best as enableDescriptorBuffer()/enableDescriptorImage().

Reply:
Right, I came to nearly the same conclusion but decided at the time that shorter method names would be better. Note that the word "descriptor" shows up twice in `DescriptorConfigurator::enableDescriptorBuffer()` so why not just make it `DescriptorConfigurator::enableBuffer()`?

But really I could have gone either way. `DescriptorConfigurator::enableDescriptorBuffer()` is not too bad in length.

Reply:
Also, I want to keep this PR to the uniform/storage issue and follow-up with a rename for the texture/image methods right after.

Reply:
Perhaps just DescriptorConfigurator::enable(..) then?

Though, as we'll need to keep it consistent with the GraphicsPipelineConfigurator API.

Reply:
If you can stand the slight inconsistency for this PR, I'd like to make this change first, keeping it as is at the moment (`enableDescriptorBuffer()`, etc) and come back with another PR to rename `enableTexture()` to `enableDescriptorImage()`. This is because the down-stream changes are many and separating this out keeps it all straight in my head.